### ISSUE TITLE
End2endTest failed on Travis

### ISSUE THREAD

**Issue by murgatroid99** [grpc/grpc#1529]
This is the error log

```
D0508 01:09:34.969442329   14480 test_config.c:52] test slowdown: machine=4.000000 build=1.000000 total=4.000000
[==========] Running 20 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 20 tests from End2endTest
[ RUN      ] End2endTest.SimpleRpc
D0508 01:09:35.177234746   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.SimpleRpc (301 ms)
[ RUN      ] End2endTest.MultipleRpcs
D0508 01:09:35.470927352   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.MultipleRpcs (301 ms)
[ RUN      ] End2endTest.RpcDeadlineExpires
D0508 01:09:35.672564820   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
E0508 01:09:35.673198316   14517 tcp_client_posix.c:149] socket error: 104
D0508 01:09:35.673295260   14480 iomgr.c:107] Waiting for 0 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.RpcDeadlineExpires (201 ms)
[ RUN      ] End2endTest.RpcLongDeadline
D0508 01:09:35.873739525   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.RpcLongDeadline (202 ms)
[ RUN      ] End2endTest.EchoDeadline
D0508 01:09:36.174785024   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.EchoDeadline (300 ms)
[ RUN      ] End2endTest.EchoDeadlineForNoDeadlineRpc
D0508 01:09:36.475355203   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.EchoDeadlineForNoDeadlineRpc (301 ms)
[ RUN      ] End2endTest.UnimplementedRpc
D0508 01:09:36.775805232   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.UnimplementedRpc (300 ms)
[ RUN      ] End2endTest.RequestStreamOneRequest
D0508 01:09:37.176405053   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.RequestStreamOneRequest (401 ms)
[ RUN      ] End2endTest.RequestStreamTwoRequests
D0508 01:09:37.476572849   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.RequestStreamTwoRequests (300 ms)
[ RUN      ] End2endTest.ResponseStream
D0508 01:09:37.876984973   14480 iomgr.c:107] Waiting for 2 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.ResponseStream (400 ms)
[ RUN      ] End2endTest.BidiStream
I0508 01:09:38.077435127   14581 end2end_test.cc:146] recv msg hello0
I0508 01:09:38.077597243   14581 end2end_test.cc:146] recv msg hello1
I0508 01:09:38.077689144   14581 end2end_test.cc:146] recv msg hello2
D0508 01:09:38.177322906   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.BidiStream (301 ms)
[ RUN      ] End2endTest.DiffPackageServices
D0508 01:09:38.578101587   14480 iomgr.c:107] Waiting for 0 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.DiffPackageServices (400 ms)
[ RUN      ] End2endTest.BadCredentials
E0508 01:09:38.678200976   14480 json_token.c:106] Invalid json string 
E0508 01:09:38.678219107   14480 credentials.c:755] Invalid input for service account credentials creation
[       OK ] End2endTest.BadCredentials (1001 ms)
[ RUN      ] End2endTest.ClientCancelsRpc
D0508 01:09:39.989578282   14480 iomgr.c:107] Waiting for 2 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.ClientCancelsRpc (400 ms)
[ RUN      ] End2endTest.ServerCancelsRpc
D0508 01:09:40.379261735   14480 iomgr.c:107] Waiting for 1 iomgr objects to be destroyed and executing final callbacks
[       OK ] End2endTest.ServerCancelsRpc (400 ms)
[ RUN      ] End2endTest.ClientCancelsRequestStream
E0508 01:09:40.679387152   14690 completion_queue.c:172] assertion failed: !cc->shutdown
FAILED: bins/opt/end2end_test [ret=-6]
```

**Comment by ctiller** [comment]
This is a duplicate of #1514. The fix was #1521. Did your PR include #1521? (it was only merged to master a few hours ago).

**Comment by murgatroid99** [comment]
No, I think that test ran before that.
